lib/shopify_api/call_limit.ex

```diff
@@ -3,30 +3,56 @@ defmodule ShopifyAPI.CallLimit do
   Responsible for handling ShopifyAPI call limits in a HTTPoison.Response
   """
   @shopify_call_limit_header "X-Shopify-Shop-Api-Call-Limit"
+  @shopify_retry_after_header "Retry-After"
   @over_limit_status_code 429
   # API Overlimit error code
+  @spec limit_header_or_status_code(any) :: nil | :over_limit
   def limit_header_or_status_code(%{status_code: @over_limit_status_code()}),
     do: :over_limit
 
   def limit_header_or_status_code(%{headers: headers}),
-    do: Enum.find(headers, fn header -> elem(header, 0) == @shopify_call_limit_header end)
+    do: get_header(headers, @shopify_call_limit_header)
 
   def limit_header_or_status_code(_conn), do: nil
 
-  def get_api_call_limit(nil), do: nil
+  def get_api_remaining_calls(nil), do: 0
 
-  def get_api_call_limit(:over_limit), do: 0
+  def get_api_remaining_calls(:over_limit), do: 0
 
-  def get_api_call_limit(header) do
+  def get_api_remaining_calls(header_value) do
     # comes in the form "1/40": 1 taken of 40
-    header
-    |> get_value_from_header
+    header_value
     |> String.split("/")
     |> Enum.map(&String.to_integer/1)
     |> calculate_available
   end
 
-  defp get_value_from_header({_, value}), do: value
+  def get_retry_after_header(%{headers: headers}) do
+    get_header(headers, @shopify_retry_after_header, "2.0")
+  end
+
+  def get_retry_after_milliseconds(header_value) do
+    {seconds, remainder} = Integer.parse(header_value)
+
+    {milliseconds, ""} =
+      remainder
+      |> String.replace_prefix(".", "")
+      |> String.pad_trailing(3, "0")
+      |> Integer.parse()
+
+    seconds * 1000 + milliseconds
+  end
+
+  def get_header(headers, header_name, default \\ nil) do
```

Comment:
We should use this https://hexdocs.pm/plug/Plug.Conn.html#get_resp_header/2 instead of recreating it

Reply:
As mentioned elsewhere, that method expects a `Plug.Conn` and not an `HTTPoison.Response` and I am not finding anything in HTTPoison to handle getting headers.

Reply:
That functionality is delegated to hackney, which would suggest doing this instead:

```elixir
def get_header(headers, header_name, default \\ nil) do
  :hackney_headers_new.get_value(
    header_name,
    :hackney_headers_new.from_list(headers),
    default
  )
end
```

Yes, `hackney_headers_new`, not `hackney_headers`. The only circumstance I think this would handle better than `Enum.find_value` is if there are multiple copies of the same header in the response.